docs: Update home page (#163)
canonical/spark-k8s-bundle: docs/index.md

(index)=
# Charmed Apache Spark documentation

Charmed Apache Spark solution is a set of Canonical supported artefacts (including charms, rocks and
snaps) that makes operating Apache Spark workloads on Kubernetes seamless, secure and
production-ready.

Charmed Apache Spark solution includes the
[Client tools snap for Apache Spark](https://snapcraft.io/spark-client),
[spark8t toolkit](https://github.com/canonical/spark-k8s-toolkit-py) and other components, like

> [Charmed Apache Spark rock](https://github.com/canonical/charmed-spark-rock/pkgs/container/charmed-spark).

Reviewer: **question** shouldn't we also add the charmed operators? also I would put things in priority: charms, rock, snaps and python toolkit

Reviewer: Shall just put a direct referent that we expose a JDBC endpoint via Kyuubi? it seems one of the "main" use-cases and services that we have right now

Reviewer: ```suggestion
Charmed Apache Spark solution includes the:
* [Charmed Operators] to deploy and manage Spark-related components, such as Apache Kyuubi and Spark History Server, using Juju
* [OCI Images] that benefits from security patching and bug fixing support to run Spark securely 
* Client tools snap, either in the form of a [snap](https://snapcraft.io/spark-client) or [python package](https://github.com/canonical/spark-k8s-toolkit-py) to simplify the user experience of running Spark on K8s
```

For an overview of all components in the solution, see the
[Components overview page](explanation-component-overview).

[Apache Spark](https://spark.apache.org) is a free, open-source software project by the Apache Software Foundation.

The solution helps ops teams and administrators automate Apache Spark operations from
[Day 0 to Day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/)

> with additional capabilities, such as: replication, TLS encryption, password rotation,

Reviewer: ```suggestion
with additional capabilities, such as: user management, encryption, password rotation,
```

Reviewer: not sure what replication would be in the context of spark

| [Reference](reference-index) </br> Technical information, for example: [release notes](reference-releases-index), [system requirements](reference-requirements), and [contact information](reference-contacts). | [Explanation](explanation-index) </br> Explore and grow your understanding of key topics, such as: [security](explanation-security), [cryptography](explanation-cryptography), [solution components](explanation-component-overview), [configuration](explanation-configuration), and [monitoring](explanation-monitoring). |

## Project and community

Charmed Apache Spark is a distribution of Apache Spark. It’s an open-source project that welcomes community contributions, suggestions, fixes and constructive feedback.  

- [Read our Code of Conduct](https://ubuntu.com/community/code-of-conduct)  
- [Join the Discourse forum](https://discourse.charmhub.io/tag/spark)  

> - [Contribute](https://github.com/canonical/spark-client-snap) and report [issues](https://github.com/canonical/spark-client-snap/issues/new)  

Reviewer: ```suggestion
- [Contribute](https://github.com/canonical/spark-k8s-bundle) and report [issues](https://github.com/canonical/spark-k8s-bundle/issues/new)  
```

**todo** I would use the `spark-k8s-bundle` repo for submitting issues. I don't think we should prioritize the `spark-client-snap` here

(index)=
# Charmed Apache Spark documentation

Charmed Apache Spark solution is a set of Canonical supported artefacts (including charms, rocks and
snaps) that makes operating Apache Spark workloads on Kubernetes seamless, secure and
production-ready.

Charmed Apache Spark solution includes the:

* Charmed operators to deploy and manage Apache Spark-related components, such as [Apache Kyuubi](https://charmhub.io/kyuubi-k8s) and [Spark History Server](https://charmhub.io/spark-history-server-k8s), using [Juju](https://canonical.com/juju)
* [OCI Images](https://github.com/canonical/charmed-spark-rock/pkgs/container/charmed-spark) that benefit from security patching and bug fixing support to run Apache Spark securely 

> * Client tools snap, either in the form of a [snap](https://snapcraft.io/spark-client) or [python package](https://github.com/canonical/spark-k8s-toolkit-py) to simplify the user experience of running Apache Spark on K8s

Author: Suggestion: This `snap` seems redundant, let's remove it.

```suggestion
* Client tools, either in the form of a [snap](https://snapcraft.io/spark-client) or [python package](https://github.com/canonical/spark-k8s-toolkit-py) to simplify the user experience of running Apache Spark on K8s
```
